compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -534,9 +534,18 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
     }
   }
 
+  /** An extractor for closures with their def contained in a block. */
```

Comment:
I'd rename it to `closureDef` and return just the `def`. The closure part is used nowhere.